app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -509,6 +554,102 @@ public void onTabReselected(TabLayout.Tab tab) {
+                if (intent == null){
+                    Log.d("barcode card id editor", "barcode and card id editor picker returned without an intent");
+                    return;
+                }
+                BarcodeValues barcodeValues = Utils.parseSetBarcodeActivityResult(Utils.BARCODE_SCAN, result.getResultCode(), intent, getApplicationContext());
+
+                cardId = barcodeValues.content();
+                barcodeType = barcodeValues.format();
+                barcodeId = "";
+            }
+        });
+
+        mCropperLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+            Intent intent = result.getData();
+            if (intent == null){
+                Log.d("cropper", "ucrop returned a null intent");
+                return;
+            }
+            if (result.getResultCode() == Activity.RESULT_OK) {
+                Uri debugUri = UCrop.getOutput(intent);
+                if (debugUri == null){
+                    throw new RuntimeException("ucrop returned success but not destination uri!");
+                }
+                Log.d("cropper", "ucrop produced image at " + debugUri);
+                Bitmap bitmap = BitmapFactory.decodeFile(getCacheDir() + "/" + TEMP_CROP_IMAGE_NAME);
+
+                if (bitmap != null) {
+                    bitmap = Utils.resizeBitmap(bitmap);
+                    if (requestedFrontImage()) {
+                        mFrontImageUnsaved = true;
+                        setCardImage(cardImageFront, bitmap);
+                    } else {
+                        mBackImageUnsaved = true;
+                        setCardImage(cardImageBack, bitmap);
+                    }
+                    Log.d("cropper", "mRequestedImage: " + mRequestedImage);
+                    mCropperFinishedType = mRequestedImage;
+                    hasChanged = true;
+                } else {
+                    Toast.makeText(LoyaltyCardEditActivity.this, R.string.errorReadingImage, Toast.LENGTH_LONG).show();
+                }
+            }else if(result.getResultCode() == UCrop.RESULT_ERROR){
+                Throwable e = UCrop.getError(intent);
+                if (e == null){
+                    throw new RuntimeException("ucrop returned error state but not and error!");
+                }
+                Log.e("cropper error", e.toString());
+            }
+        });
+
+        mCropperOptions = new UCrop.Options();
+        setCropperOptions();
+        setCropperTheme();
+    }
+
+    private void setCropperOptions(){
+        mCropperOptions.setCompressionFormat(TEMP_CROP_IMAGE_FORMAT);
+        mCropperOptions.setFreeStyleCropEnabled(true);
+        mCropperOptions.setHideBottomControls(false);
+        // needed when bottom controls are hidden
+        //mCropperOptions.setAllowedGestures(UCropActivity.ALL, UCropActivity.ALL, UCropActivity.ALL);
+        mCropperOptions.setAspectRatioOptions(0,
```

Comment:
I noticed the cropper gives me 1:1 and card as aspect ratios, but it actually doesn't allow me to just select the whole image. Could you please add an "Original" size after all, as default, which just selects the whole image? So people don't have to go through extra steps dragging stuff if they don't want to crop the image they took.